dali/sasha/saxref.cpp

```diff
@@ -368,9 +367,9 @@ struct cDirDesc
             mp->init(drv,pf,node,numnodes);
             mp->next = file->misplaced;
             file->misplaced = mp;
-
+            // NB: still perform setpresent() below, so that later 'orphan' and 'found' scanning can spot the part as orphaned or part of a found file.
```

Comment:
The way it used to regard 'misplaced' (pre. HPCC-17967) meant that it only marked parts as misplaced if their part number mod. width did not match the current node # it was scanning.
This lead to only some parts of unrelated clusters and parts with part numbers that just happened to mod. to the current node to be regarded as !misplaced.
As consequence they were included in other later scans erroneously.

The misplaced flag now flags any part that is not strictly within bounds of the cluster group being scanned.
The above 'setpresent' line ensures that the misplaced part is still flagged as present within the per global file structure, so that later orphan and 'lost' scanning can spot that it is an orphaned file part or part of a 'lost' file.
NB: Unrelated to this change or the regression, prior parts marked as misplaced might not be spotted as orphans, and would be completely ignored (not seen in any report).